backend-rust/src/indexer.rs

```diff
@@ -876,7 +915,182 @@ WHERE blocks.finalization_time IS NULL AND blocks.height <= last.height
+                foundation_account,
+                baker_reward,
+                foundation_charge,
+                ..
+            } => vec![
+                PreparedBlockSpecialEvent {
+                    account_address: foundation_account.to_string(),
+                    amount: foundation_charge.micro_ccd.try_into()?,
+                    block_height: block_height.height.try_into()?,
+                    slot_time,
+                    transaction_type: AccountStatementEntryType::FoundationReward,
+                },
+                PreparedBlockSpecialEvent {
+                    account_address: baker.to_string(),
+                    amount: baker_reward.micro_ccd.try_into()?,
+                    block_height: block_height.height.try_into()?,
+                    slot_time,
+                    transaction_type: AccountStatementEntryType::BakerReward,
+                },
+            ],
+            SpecialTransactionOutcome::PaydayFoundationReward {
+                foundation_account,
+                development_charge,
+            } => vec![PreparedBlockSpecialEvent {
+                account_address: foundation_account.to_string(),
+                amount: development_charge.micro_ccd.try_into()?,
+                block_height: block_height.height.try_into()?,
+                slot_time,
+                transaction_type: AccountStatementEntryType::FoundationReward,
+            }],
+            SpecialTransactionOutcome::PaydayAccountReward {
+                account,
+                transaction_fees,
+                baker_reward,
+                finalization_reward,
+            } => vec![
+                PreparedBlockSpecialEvent {
+                    account_address: account.to_string(),
+                    amount: transaction_fees.micro_ccd.try_into()?,
+                    block_height: block_height.height.try_into()?,
+                    slot_time,
+                    transaction_type: AccountStatementEntryType::TransactionFeeReward,
+                },
+                PreparedBlockSpecialEvent {
+                    account_address: account.to_string(),
+                    amount: baker_reward.micro_ccd.try_into()?,
+                    block_height: block_height.height.try_into()?,
+                    slot_time,
+                    transaction_type: AccountStatementEntryType::BakerReward,
+                },
+                PreparedBlockSpecialEvent {
+                    account_address: account.to_string(),
+                    amount: finalization_reward.micro_ccd.try_into()?,
+                    block_height: block_height.height.try_into()?,
+                    slot_time,
+                    transaction_type: AccountStatementEntryType::FinalizationReward,
+                },
+            ],
+            SpecialTransactionOutcome::BlockAccrueReward {
+                ..
+            }
+            | SpecialTransactionOutcome::PaydayPoolReward {
+                ..
+            } => Vec::new(),
```

Comment:
I am surprised these are not implemented as these also include rewards for accounts

Reply:
Lets implement that

Reply:
There seem to be some issue here. 

This was what we did on the old CCDScan, however traversing the database we are getting these into the database. 

I where those stem from and why we differ on those.

Reply:
As this deviates from the existing flow I would like to wait with this. I added a TODO, and will do that after I have validated correctness between the two instances of CCDScan.

Reply:
Pretty sure these outcomes were introduced in a later protocol version, so this might be why the code for handling it is scattered in the old backend.